src/three-idx.ts

```diff
@@ -92,37 +93,37 @@ export class ThreeIDX {
    */
   async createIDX(authEntry?: NewAuthEntry): Promise<void> {
     const entry = authEntry ? await this.createAuthMapEntry(authEntry) : {}
-    await this.createKeychainDoc(entry)
-    this.docs.idx = await this.ceramic.createDocument('tile', {
-      metadata: { controllers: [this.id], schema: IdentityIndex },
-      content: { [KEYCHAIN_DEF]: this.docs[KEYCHAIN_DEF].id.toUrl('base36') },
-    })
-    await this.docs.threeId.change({
-      content: Object.assign(this.docs.threeId.content, {
-        idx: this.docs.idx.id.baseID.toUrl('base36'),
-      }),
-    })
-    await this.pinAllDocs()
+    // eslint-disable-next-line prettier/prettier
+    await Promise.all([
+      this.loadKeychainDoc(this.id),
+      this.loadIDXDoc(this.id)
+    ])
+    // eslint-disable-next-line prettier/prettier
+    await Promise.all([
+      this.pinAllDocs(),
+      this.updateKeychainDoc(entry),
+      this.addKeychainToIDX()
+    ])
   }
 
   /**
    * Returns the encrypted JWE for the given authLink
    */
   async loadIDX(authLink: string): Promise<EncKeyMaterial | null> {
     this.docs[authLink] = await this.ceramic.createDocument(
-      'account-link',
+      'caip10-link',
       { metadata: { controllers: [authLink] } },
-      { applyOnly: true }
+      { anchor: false, publish: false }
     )
     const did = this.docs[authLink].content
     if (!did) return null
-    this.docs.threeId = await this.ceramic.loadDocument(`ceramic://${did.split(':')[2] as string}`)
-    const idxDocid = this.docs.threeId.content.idx
-    if (!idxDocid) return null
-    this.docs.idx = await this.ceramic.loadDocument(idxDocid)
-    const authKeychainDocid = this.docs.idx.content[KEYCHAIN_DEF]
-    if (!authKeychainDocid) return null
-    this.docs[KEYCHAIN_DEF] = await this.ceramic.loadDocument(authKeychainDocid)
+    await Promise.all([
```

Comment:
I find it a bit strange to have this IIF in the `Promise.all()` call, maybe this could be done like this instead?
```ts
const [threeId] = await Promise.all([
  this.ceramic.loadDocument(did.split(':')[2] as string),
  this.loadKeychainDoc(did),
  this.loadIDXDoc(did),
])
this.docs.threeId = threeId
```

Reply:
Nice!